src/main/java/edu/wpi/grip/core/events/StepMovedEvent.java

```diff
@@ -0,0 +1,37 @@
+package edu.wpi.grip.core.events;
+
+import com.google.common.base.MoreObjects;
+import edu.wpi.grip.core.Step;
+
+/**
+ * An event that occurs when a new step is moved from one position to another in the pipeline
+ */
+public class StepMovedEvent {
+    private final Step step;
+    private final int distance;
+
+    /**
+     * @param step     The step being moved
+     * @param distance The number of indices (positive or negative) to move the step by
+     */
+    public StepMovedEvent(Step step, int distance) {
```

Comment:
Because steps don't know what index they're at, and a distance better maps to what the buttons actually do